src/test/java/org/assertj/core/util/TestDeepEquals.java

```diff
@@ -0,0 +1,375 @@
+package org.assertj.core.util;
+
+import static java.lang.Math.E;
+import static java.lang.Math.PI;
+import static java.lang.Math.atan;
+import static java.lang.Math.cos;
+import static java.lang.Math.log;
+import static java.lang.Math.pow;
+import static java.lang.Math.sin;
+import static java.lang.Math.tan;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.SortedSet;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentSkipListMap;
+
+import org.junit.Test;
+
+/**
+ * @author John DeRegnaucourt
+ * @author sapradhan8
+ * @author Pascal Schumacher
+ */
+public class TestDeepEquals {
```

Comment:
let's use AssertJ assertions by adding `assertAreDeepEquals(x,y)` that simply calls `assertThat(DeepEquals.deepEquals(x, y)).isTrue()`